### PR TITLE
chore: add test to demonstrate trace exporter constructor's ability t…

### PR DESCRIPTION
**Pull request by jonahrosenblum**
**Description**
- Adds a test case to demonstrate how the trace exporter can get the correct project-id while in a GCE environment even when no arguments are supplied.
- Update documentation and examples to reflect this behavior.

**Testing**
- Does not interfere with other unit tests
- Confirmed behavior on Cloud Functions

**Relevant Info**
- Addresses issue #98
- Addresses issue #106

### CODE REVIEW

**Comment by jonahrosenblum** [comment]
I apologize for not running the linter prior to making a pull request. I have been unable to get it working locally, it looks like I need to remove/insert a blank line?

**Comment by mayurkale22** [comment]
@dancalvert-google @kintel FYI. I am not sure who else to add to review this PR. This PR is pending since long time.

File: packages/opentelemetry-cloud-trace-exporter/package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "@opentelemetry/core": "^0.7.0",
     "@opentelemetry/resources": "^0.7.0",
     "@opentelemetry/tracing": "^0.7.0",
+    "gcp-metadata": "^4.1.0",
```
Comment (kintel) [inline packages/opentelemetry-cloud-trace-exporter/package.json:59]: My understanding is that the `gcp-metadata` package is only used by the tests, in which case it should be part of "devDependencies".

File: packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts [hunk 2]
```diff
@@ -42,6 +54,20 @@ describe('Google Cloud Trace Exporter', () => {
         assert.deepStrictEqual(id, 'not-real');
       });
     });
+
+    it('should construct exporter in GCE environment without args', async () => {
```
Comment (kintel) [inline packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:58]: s/GCE/GCP/ - the metadata server (even though it's called the GCE Metadata server) is available in all/most GCP compute products.

File: samples/trace/index.js [hunk 3]
```diff
@@ -28,19 +28,11 @@ const { TraceExporter } = require('@google-cloud/opentelemetry-cloud-trace-expor
...
-// and not this code itself. Checking for existence here but not retaining (as not needed)
-if (!projectId || !process.env.GOOGLE_APPLICATION_CREDENTIALS) {
-  throw Error('Unable to proceed without a Project ID');
-}
-
 const provider = new NodeTracerProvider();
 
-// Initialize the exporter
-const exporter = new TraceExporter({projectId: projectId});
+// Initialize the exporter. When run in the Google Cloud Environment no
```
Comment (kintel) [inline samples/trace/index.js:33]: s/Google Cloud Environment/GCP/ - "Google Cloud Environment" is not a well-defined concept.

File: packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts [hunk 4]
```diff
@@ -42,6 +54,20 @@ describe('Google Cloud Trace Exporter', () => {
...
+      delete process.env.GCLOUD_PROJECT;
```
Comment (kintel) [inline packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:59]: Why delete this variable in particular? There are a few variables historically used to define a project ID. See here for how the google auth library actually picks this up: https://github.com/googleapis/google-auth-library-nodejs/blob/88a46b0959e3abb3b784f1fcb7fb4cff4272a0f6/src/auth/googleauth.ts#L575

**Comment by kintel** [comment]
@mayurkale22 I expect you're the only other available reviewer at the time beeing.

File: packages/opentelemetry-cloud-trace-exporter/package.json (hunk 1 above)
Comment (mayurkale22) [reply packages/opentelemetry-cloud-trace-exporter/package.json:59]: +1

File: packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts [hunk 5]
```diff
@@ -42,6 +54,20 @@ describe('Google Cloud Trace Exporter', () => {
...
+      const gcpMock = nock(HOST_ADDRESS)
+        .get(PROJECT_ID_PATH)
+        .reply(200, () => 'not-real', HEADERS);
+      const exporter = new TraceExporter({});
```
Comment (mayurkale22) [inline packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:63]:
```suggestion
      const exporter = new TraceExporter();
```

File: samples/trace/index.js (hunk 3 above)
Comment (mayurkale22) [reply samples/trace/index.js:33]: I would say: when your application is running on Google Cloud, you don't need to provide authentication credentials or project id.

File: packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts (hunk 4 above)
Comment (jonahrosenblum) [reply packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:59]: If this variable is present the mock metadata server is not needed, without it the metadata server is needed. This was what I deduced from testing a month ago but I could try it again to verify that this is the only variable defined in this environment.

Comment (jonahrosenblum) [reply packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:59]: https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/blob/master/packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts#L28

Deleting this variable is in response to how the testing environment sets it in the testing environment. I ran the test again without the metadata server and confirmed that none of the other potential environment variables are present.

Comment (kintel) [reply packages/opentelemetry-cloud-trace-exporter/test/exporter.test.ts:59]: Sounds good, I would still add a comment explaining that this is set by the test env and that you need to undefine it to force a metadata server request.

**Comment by jonahrosenblum** [comment]
@kintel @mayurkale22 Let me know if there is anything else I should clarify or fix. Thank you for all the feedback!

File: packages/opentelemetry-cloud-trace-exporter/package.json [hunk 6]
```diff
@@ -38,17 +38,18 @@
...
     "codecov": "3.7.2",
     "gts": "2.0.2",
     "mocha": "7.2.0",
     "nyc": "15.1.0",
     "rimraf": "3.0.2",
     "sinon": "9.0.2",
     "ts-mocha": "7.0.0",
-    "typescript": "3.9.7"
+    "typescript": "3.9.7",
+    "gcp-metadata": "^4.1.0"
```
Comment (mayurkale22) [inline packages/opentelemetry-cloud-trace-exporter/package.json:52]: nit: usually we try to list dependencies in the alphabetical order. I would move `gcp-metadata` just above `gts` and `nock` just after `mocha`.

Comment (jonahrosenblum) [reply packages/opentelemetry-cloud-trace-exporter/package.json:52]: Makes sense, thanks for the catch.

**Comment by mayurkale22** [comment]
@jonahrosenblum This is good to go, could you please fix the build

**Comment by jonahrosenblum** [comment]
> @jonahrosenblum This is good to go, could you please fix the build

@mayurkale22 Fixed.

**Pull request merged**